datafusion-cli/src/functions.rs

```diff
@@ -196,3 +209,110 @@ pub fn display_all_functions() -> Result<()> {
     println!("{}", pretty_format_batches(&[batch]).unwrap());
     Ok(())
 }
+
+/// PARQUET_META table function
+struct ParquetMetadataTable {
+    schema: SchemaRef,
+    batch: RecordBatch,
+}
+
+#[async_trait]
+impl TableProvider for ParquetMetadataTable {
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+
+    fn schema(&self) -> arrow::datatypes::SchemaRef {
+        self.schema.clone()
+    }
+
+    fn table_type(&self) -> datafusion::logical_expr::TableType {
+        datafusion::logical_expr::TableType::Base
+    }
+
+    async fn scan(
+        &self,
+        _state: &SessionState,
+        projection: Option<&Vec<usize>>,
+        _filters: &[Expr],
+        _limit: Option<usize>,
+    ) -> Result<Arc<dyn ExecutionPlan>> {
+        Ok(Arc::new(MemoryExec::try_new(
+            &[vec![self.batch.clone()]],
+            TableProvider::schema(self),
+            projection.cloned(),
+        )?))
+    }
+}
+
+pub struct ParquetMetadataFunc {}
+
+impl TableFunctionImpl for ParquetMetadataFunc {
+    fn call(&self, exprs: &[Expr]) -> Result<Arc<dyn TableProvider>> {
+        let Some(Expr::Column(Column { name, .. })) = exprs.get(0) else {
+            return plan_err!("read_csv requires at least one string argument");
+        };
+
+        let file = File::open(name)?;
+        let reader = SerializedFileReader::new(file)?;
+        let metadata = reader.metadata();
+
+        let schema = Arc::new(Schema::new(vec![
+            Field::new("version", DataType::Int32, false),
+            Field::new("num_rows", DataType::Int64, false),
+            Field::new("created_by", DataType::Utf8, false),
+            Field::new("columns_order", DataType::Utf8, false),
+            Field::new("num_row_groups", DataType::Int64, false),
+            Field::new("row_groups", DataType::Utf8, false),
```

Comment:
Perhaps this field being a list of structs would be more precise?

Reply:
Yeah, agree with you!  a list of structs would be more precise! I implement the first version schema using `List(Struct)`.

But if the list contains lots of row_groups (100+ or more), then all 100+ rows groups will be displayed in one cell. 
  Then the display may look not so user-friendly?

Reply:
If there are many row_groups, wouldn't the single String also fill up the single cell?

Reply:
each row group will be one row